File: Python/Medium/الترتيب حسب الطول.py

# Creator: OmarAfet
# https://profile.satr.codes/OmarAfet/public/overview
# https://github.com/OmarAfet

def sortByLength(txt: str) -> str:
    words = txt.split()

    sorted_words = sorted(words, key=lambda x: (len(x), x))

    return ' '.join(sorted_words)

File: challenges/JavaScript/2. Medium/الترتيب حسب الطول.js
/*

Creator: OmarAfet
https://profile.satr.codes/OmarAfet/public/overview
https://github.com/OmarAfet

*/

function sortByLength(txt) {
	const words = txt.split(" ");
	const sortedWords = words.sort((a, b) => {
		if (a.length !== b.length) {
			return a.length - b.length;
		}
		return a.localeCompare(b);
	});
	return sortedWords.join(" ");
}
